lib/palindrome_check.rb

```diff
@@ -1,7 +1,21 @@
-# A method to check if the input string is a palindrome.
-# Return true if the string is a palindrome. Return false otherwise.
-# Time complexity: ?
-# Space complexity: ?
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  if my_phrase == nil
+    return false
+  end
+  i = 0
+  j = my_phrase.length - 1
+  while i < j
```

Comment:
**Interview tip:** make sure to _explicitly_ check for "off-by-one" errors in code like this, especially when you're writing code on a whiteboard.

(The tests already do this, so no need to do it yourself for this problem. :smile:)